### DISCUSSION TITLE
How to Create Private Releases?

### DISCUSSION THREAD

**Discussion by GiovanniYuriMita** [hystax/optscale#384] [Q&A] [answered]
# How to Create Private Releases?

I've cloned the repository and made several modifications to the UI, including the addition of dark mode and some minor personal adjustments. Now, I'd like to create a private release of my customized version to deploy on a VM for version updates.

Could someone guide me through the process of creating tags and releases? 

Any detailed steps or resources would be greatly appreciated. Thanks in advance!

**Comment by stanfra** [comment, marked as answer]
Hi @GiovanniYuriMita , we create releases manually so if you want create release go to /releases page in your repo and create new one.
To build images with custom tag you need to run command inside your optscale repo:
`bash -x build.sh . <custom-version-tag>`

**Reply by GiovanniYuriMita** [reply]
Hi, @stanfra!

Thanks for your help. I successfully built the ngui image and pushed it to my Docker Hub repository. However, I'm still a bit unclear about updating it using the runkube-py script. Since I only modified ngui, I'm unsure if running something like `./runkube.py --with-elk --update-only -- <deployment name> <my-custom-version>` would attempt to update all deployment images to `<my-custom-version>`, which, in theory, would break the other deployments.

Can you guide me in this matter?

**Reply by stanfra** [reply]
If you want to make release you need to build all images with custom tag and push them to Docker Hub to use runkube, if you push only ngui image you can pull it to your cluster using "docker pull" command, then tag it as local and restart ngui pod to restart it from new local image

**Reply by GiovanniYuriMita** [reply]
Thank you so much!! I got it now!

It also gave me an idea of how I can create a new setup and deployment method to test on eks on the future.

**Comment by tm-hystax** [comment]
By the way, OptScale has the ability to customize colors and fonts within the organization. Open page ```https://<cluster_ip>/theme-settings``` for customization.
![image](https://github.com/user-attachments/assets/b2037cf3-4c23-4519-a80a-0160e5b4b82d)

![image](https://github.com/user-attachments/assets/4ae79317-111d-4c2b-baf2-fd2027ec4b21)

**Reply by GiovanniYuriMita** [reply]
That's awesome, @tm-hystax! 
Thank you for sharing!
